server/src-lib/Hasura/GraphQL/Schema.hs

```diff
@@ -64,59 +64,29 @@ buildGQLContext
 buildGQLContext =
   proc (queryType, allTables, allFunctions, allRemoteSchemas, allActions, nonObjectCustomTypes) -> do
 
-    -- Scroll down a few pages for the actual body...
+    SQLGenCtx{ stringifyNum } <- bindA -< askSQLGenCtx
 
     let allRoles = Set.insert adminRoleName $
              (allTables ^.. folded.tiRolePermInfoMap.to Map.keys.folded)
           <> (allActionInfos ^.. folded.aiPermissions.to Map.keys.folded)
-
-        tableFilter    = not . isSystemDefined . _tciSystemDefined
-        functionFilter = not . isSystemDefined . fiSystemDefined
-
-        graphQLTableFilter tableName tableInfo =
-          -- either the table name should be GraphQL compliant
-          -- or it should have a GraphQL custom name set with it
-          isGraphQLCompliantTableName tableName
-          || (isJust . _tcCustomName . _tciCustomConfig . _tiCoreInfo $ tableInfo)
-
-        validTables = Map.filter (tableFilter . _tiCoreInfo) allTables
-        -- Only tables that have GraphQL compliant names will be added to the schema.
-        -- We allow tables which don't have GraphQL compliant names, so that RQL CRUD
-        -- operations can be performed on them
-        graphQLTables = Map.filterWithKey graphQLTableFilter validTables
-        validFunctions = Map.elems $ Map.filter functionFilter allFunctions
-
         allActionInfos = Map.elems allActions
         queryRemotesMap =
           fmap (map fDefinition . piQuery . rscParsed . fst) allRemoteSchemas
-        buildFullestDBSchema
-          :: m ( Parser 'Output (P.ParseT Identity) (OMap.InsOrdHashMap G.Name (QueryRootField UnpreparedValue))
-               , Maybe (Parser 'Output (P.ParseT Identity) (OMap.InsOrdHashMap G.Name (MutationRootField UnpreparedValue)))
-               )
-        buildFullestDBSchema = do
-          SQLGenCtx{ stringifyNum } <- askSQLGenCtx
-          let gqlContext =
-                (,)
-                <$> queryWithIntrospection (Set.fromMap $ graphQLTables $> ())
-                      validFunctions mempty mempty
-                      allActionInfos nonObjectCustomTypes
-                <*> mutation (Set.fromMap $ graphQLTables $> ()) mempty
-                      allActionInfos nonObjectCustomTypes
-          flip runReaderT (adminRoleName, graphQLTables, Frontend, QueryContext stringifyNum queryType queryRemotesMap) $
-            P.runSchemaT gqlContext
-
-    -- build the admin context so that we can check against name clashes with remotes
-    adminHasuraContext <- bindA -< buildFullestDBSchema
+        queryContext = QueryContext stringifyNum queryType queryRemotesMap
+
+    -- build the admin DB-only context so that we can check against name clashes with remotes
```

Comment:
```suggestion
    -- build the admin DB-only context so that we can check against name clashes with remotes
    -- TODO: Is there a better way to check for conflicts without actually building the admin schema?
```